### PR TITLE
Enabling OBS training-aware

### PR DESCRIPTION
**Pull request by abhinavnmagic**

### CODE REVIEW

**Comment by dbogunowicz** [comment]
@abhinavnmagic Please check whether my recent changes are working correctly, by running export script e.g.
```bash
sparseml.ultralytics.export --one-shot {recipe_with_obs}.yaml --save_one_shot_torch True --model yolov8s.pt
```

File: src/sparseml/yolov8/trainers.py [hunk 1]
```diff
@@ -571,7 +604,20 @@ def export(self, **kwargs):
             for p in self.model.parameters():
                 p.requires_grad = True
             manager = ScheduledModifierManager.from_yaml(one_shot)
-            manager.apply(self.model)
+
+            # TODO: I hate the fact that to scale this functionality to other tasks
+            # we might have to (worst case scenario) copy and paste many
+            # arguments from the train.py. This is not ideal.
```
Comment (abhinavnmagic) [inline src/sparseml/yolov8/trainers.py:610]: Can we not just arguments for data and device to export CLI?

File: src/sparseml/yolov8/trainers.py [hunk 2]
```diff
@@ -571,7 +604,20 @@ def export(self, **kwargs):
...
+            overrides = self.overrides.copy()
+            overrides["data"] = "coco128.yaml"  # args["data"]
+            overrides["device"] = "cpu"
+            trainer = self.TrainerClass(overrides=overrides)
+
+            manager.apply(
+                self.model,
+                # maybe we could check whether OBS pruner is in the manager?
```
Comment (abhinavnmagic) [inline src/sparseml/yolov8/trainers.py:618]: Manager does have OBS Pruner, as part of `manager.pruning_modifiers`.

File: src/sparseml/yolov8/trainers.py (hunk 1 above)
Comment (abhinavnmagic) [reply src/sparseml/yolov8/trainers.py:610]: I guess that's what you mean by copy and paste? Yes, we would essentially have to maintain two pathways: one-shot and training-aware with both of them having similar support.

**Pull request closed without merging**